### PR TITLE
fixed form builder when using 'for' option

### LINKED ISSUE #3579: form bug for has_one association
Hello, there is a problem in form builder when rendering nested fields before main . https://github.com/activeadmin/activeadmin/pull/3555#issuecomment-61483238 
Example 
```
f.inputs  for: :option do  |o|
                     o.input :option1
                     o.input :option2
                     o.input :option3  #<<<only last input is displayed
                end
 f.inputs do
      f.input :fname
      f.input :lname
    end
```

in this case only last input for option is displayed. However if I change inputs vice versa it works.

```
 f.inputs do
      f.input :fname
      f.input :lname
    end
f.inputs  for: :option do  |o|
                     o.input :option1
                     o.input :option2
                     o.input :option3  
                end

```

### PR DESCRIPTION
**Pull request by Fivell**
- fixes #3579
- DRY form builder specs

### CODE REVIEW

File: lib/active_admin/views/components/active_admin_form.rb [hunk 1]
```diff
@@ -42,11 +42,13 @@ def build(resource, options = {}, &block)
       end
 
       def inputs(*args, &block)
+        if block_given?
+          form_builder.template.assign('has_many_block'=> true)
```
Comment (timoschilling) [inline lib/active_admin/views/components/active_admin_form.rb]: Maybe this works:
```ruby
form_builder.template.assign(has_many_block: true)
```

**Comment by timoschilling** [comment]
:+1: please apply the remaining comments

**Comment by Fivell** [comment]
@timoschilling , 'has_many_block' now is symbol, let's see what travis says

**Comment by timoschilling** [comment]
can you squash the commits please

**Comment by Fivell** [comment]
@timoschilling , done

**Comment by timoschilling** [comment]
@Fivell :+1: maybe you can help with some of the other form bugs

**Pull request merged**